marshal/consumer.go

```diff
@@ -375,6 +394,35 @@ func (c *Consumer) claimTopics() {
 		}
 	}
 
+	// let's compare the new topic claims vs old ones. Upon change, we should trigger an update
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	changed := false
+	// check for missing topic claims
+	for topic := range c.claimedTopics {
+		if !latestClaims[topic] {
+			changed = true
+			break
+		}
+	}
+	// check for new topic claims
+	if !changed && len(latestClaims) != len(c.claimedTopics) {
+		changed = true
+	}
+
+	if changed {
+		c.lock.RUnlock()
```

Comment:
I agree...I think the anonymous closure can be its own named function...sure. It was small enough that I did it this way.
I was referring to running the closure asynchronously. In operations such as Terminate, which eventually calls `updateTopicClaims`, doing the update asynchronously is bad because we need to close the channel and we won't know when its done (unless we add extra synchronization).